treezigzagtraversal/zigzag.java

```diff
@@ -0,0 +1,77 @@
+import java.util.*;
+
+/**
+ * zigzag
+ */
+// sample input
+// 10 true 20 true 40 false false true 50 false false true 30 true 60 false
+// false true 73 false false
+// sample output
+// 10 30 20 40 50 60 73
```

Comment:
Either use single line comments or multi line comment here.
Also, sample input and output should be at the EOF.